drivers/interrupt_controller/intc_nuclei_eclic.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Nuclie's Extended Core Interrupt Controller
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <sys/util.h>
+#include <init.h>
+#include <soc.h>
+
+#include <sw_isr_table.h>
+
+union CLICCFG {
+	struct {
+		uint8_t _reserved0 : 1;
+		uint8_t nlbits : 4;        /*!< number of interrupt level bits */
+		uint8_t _reserved1 : 2;
+		uint8_t _reserved2 : 1;
```

Comment:
if documenting, document all fields using this style:
```c
...
/** Number of interrupt level bits */
uint8_t nlbits : 4;
...
```

(applies to other comments)

Reply:
I got it.
Checking the entire sources.